cli/src/main/scala/com/advancedtelematic/tuf/cli/Delegations.scala

```diff
@@ -49,11 +49,21 @@ object Delegations {
     }
   }
 
-  def pull(reposerverClient: ReposerverClient, delegationName: DelegatedRoleName, outputStream: OutputStream)
+  def pull(reposerverClient: ReposerverClient, delegationName: DelegatedRoleName, output: WriteOutput)
           (implicit ec: ExecutionContext): Future[SignedPayload[TargetsRole]] = {
     reposerverClient.pullDelegation(delegationName).map { delegation =>
-      outputStream.write(delegation.asJson.spaces2.getBytes)
+      output.write(delegation.asJson.spaces2.getBytes)
       delegation
     }
   }
+
+  def addTarget(input: Path, output: WriteOutput, targetFilename: TargetFilename, targetItem: ClientTargetItem): Try[Unit] = {
+    read(input).flatMap { json =>
+      json.as[TargetsRole].toTry.map(json -> _)
+    }.map { case (rawJson, targets) =>
+      val newTargets = targets.copy(targets = targets.targets + (targetFilename -> targetItem))
```

Comment:
Why is that not `val newTargets = targetFilename -> targetItem`?

Reply:
That would overwrite existing targets ?

Reply:
If I read that correctly that overwrites the current targets attribute of targets, which might contain other targets. you need to add `targetFilename -> targetItem` to targets.targets.

Reply:
I don't think so, it's kind of the point of a deep merge to not do that.

Reply:
The deep merge is for the other attributes that the user might have added to the json file that are not parsed by our encoders. I don't think the deep merge would fix this, you are overwritting an attribute directly, saying it should be this, it wont be merged.

Reply:
the map gets converted to an object, so it works:
```
val t1 = Json.obj("targets" -> Json.obj("file1" -> Json.True, "file2" -> Json.False))
val t2 = Json.obj("targets" -> Json.obj("file2" -> Json.True))
t1.deepMerge(t2)
res0: io.circe.Json =
{
"targets" : {
"file2" : true,
"file1" : true
}
}
```

Reply:
Yes, I am wrong. I still think `val newTargets = targets.copy(targets = targets.targets + (targetFilename -> targetItem))` is clearer on the intention than using the deepMerge for that. The deepMerge was meant for attributes that are not in the codecs. I will add some tests to make sure this does what I want it to do.